openerp/addons/base/ir/ir_translation.py

```diff
@@ -125,10 +125,11 @@ def finish(self):
             AND irt.type = ti.type
             AND irt.module = ti.module
             AND irt.name = ti.name
-            AND (ti.type IN ('field', 'help') OR irt.src = ti.src)
-            AND (    ti.type NOT IN ('model', 'view')
+            AND (
+                    (ti.type NOT IN ('model', 'view', 'field', 'help'))
                  OR (ti.type = 'model' AND ti.res_id = irt.res_id)
-                 OR (ti.type = 'view' AND (irt.res_id IS NULL OR ti.res_id = irt.res_id))
+                 OR (ti.type = 'view' AND (irt.res_id IS NULL OR ti.res_id = irt.res_id)
+                 OR (irt.src = ti.src))
                 )
         """
```

Comment:
make the condition even more explicit:

    AND (   (ti.type = 'model' AND ti.res_id = irt.res_id)
         OR (ti.type = 'view' AND (irt.res_id IS NULL OR ti.res_id = irt.res_id) AND irt.src = ti.src)
         OR (ti.type = 'field')
         OR (ti.type = 'help')
         OR (ti.type NOT IN ('model', 'view', 'field', 'help') AND irt.src = ti.src)
    )